code/modules/tables/interactions.dm

```diff
@@ -57,6 +57,64 @@
 			return 1
 	return 1
 
+/obj/structure/table/Crossed(var/atom/movable/am as mob|obj)
+	..()
+	if(ishuman(am))
+		var/mob/living/carbon/human/H = am
+		if(H.a_intent != I_HELP || H.m_intent == "run")
+			throw_things(H)
+		else if(H.is_diona() || H.species.get_bodytype() == "Heavy Machine")
+			throw_things(H)
+	else if((isliving(am) && !issmall(am)) || isslime(am))
+		throw_things(am)
+
+/obj/structure/table/proc/throw_things(var/mob/living/user)
+	var/list/targets = list(get_step(src,dir),get_step(src,turn(dir, 45)),get_step(src,turn(dir, -45)))
+	var/turf/T = get_turf(src)
+	var/anything_moved = FALSE
+	for (var/obj/item/I in T)
+		if (I.simulated && !I.anchored)
+			INVOKE_ASYNC(I, /atom/movable/.proc/throw_at, pick(targets), 1, 1)
+			anything_moved = TRUE
+		CHECK_TICK
+
+	if (user && anything_moved)
+		visible_message("<span class='notice'>[user] kicks everything off [src].</span>")

```

Comment:
This should probably be:
```dm
user.visible_message("<span class='notice'>[user] kicks everything off [src].</span>")
```
(Adding a self message would be a nice bonus, but isn't required)